zipkin/src/main/java/zipkin2/Span.java

```diff
@@ -635,6 +635,7 @@ public static String normalizeTraceId(String traceId) {
     if (length > 32) throw new IllegalArgumentException("traceId.length > 32");
     int zeros = validateHexAndReturnZeroPrefix(traceId);
     if (zeros == length) throw new IllegalArgumentException("traceId is all zeros");
+    if (length == 15) throw new RuntimeException("WTF");
```

Comment:
@adriancole can you provide us more context about this change? We were implicitly relying on the previous logic that left-padded 0s for trace ids generated by a legacy application. We have a workaround in place to fix the broken pipeline, but we wanted to hear why this explicitly checks for length of 15 (vs 14, 13 etc...)

Reply:
@codefromthecrypt Why was this change added? Any other hex code (eg of length 14) will be nicely prepended with a leading zero, but why should the 15 throw an exception exactly?

Reply:
guess by the exception name that this was an accident especially as there's logic later to handle padding

Reply:
I'll create a PR to remove this change then

Reply:
https://github.com/openzipkin/zipkin/pull/3471 has been created. In case something is off with it, let's continue any discussion there.

Thanks for the quick replies!